### PR TITLE
jupyter: handle heartbeat messages

### PR DESCRIPTION
**Pull request by sbinet**

### CODE REVIEW

File: jupyter/jupyter.go [hunk 1]
```diff
@@ -240,6 +268,43 @@ func (s *server) iopubHandler(c *conn) {
 	}
 }
 
+func (s *server) heartbeatHandler(c *conn) {
+	if debug {
+		log.Printf("heartbeat handler started\n")
+	}
+
+	ctx, cancel := context.WithCancel(s.ctx)
```
Comment (crawshaw) [inline jupyter/jupyter.go]: Is this cancellation context doing anything?

Comment (sbinet) [reply jupyter/jupyter.go]: not yet. (sorry, I forgot it when I crafted the CL)
but I have a CL pending that will call it when the kernel receives `"shutdown_request"`.

**Comment by sbinet** [comment]
I am not completely sure the heartbeats are being sent, but I did see the jupyter-notebook create the connection.

**Comment by sbinet** [comment]
ha! https://github.com/JuliaLang/IJulia.jl/issues/489#issuecomment-270376721

and that's matched with this: https://github.com/jupyter/jupyter_client/blob/master/jupyter_client/channels.py#L31

**Comment by crawshaw** [comment]
Huh. Not the easiest to implement protocol I've ever seen. (Documentation on what kind of sockets ZMTP creates was also in very short supply. I figured it out with trial, error, and strace.)

**Comment by sbinet** [comment]
yeah... elsewhere, I was joking (half joking?) that we should just reimplement the whole jupyter notebook server in Go, calling it "saturn: it's smaller than jupyter but has a nice ring to it", with a proper definition of messages using protobuf and gRPC (except that AFAICT, there is no router/dealer/pub/sub/... handling with gRPC).
then I realized it was also work :)

as this PR stands, it's just adding dead code. untested even.
I'd be ok to withdraw it and move on to implementing shutdown and (trying to implement) displaying images.

**Comment by crawshaw** [comment]
Yeah let's skip this for now, we can revisit it when heartbeats appear.

Displaying images sounds way more interesting.

**Pull request closed without merging**